core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -154,21 +155,33 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     Path tempMetadataFile = metadataPath(UUID.randomUUID().toString() + fileExtension);
     TableMetadataParser.write(metadata, io().newOutputFile(tempMetadataFile.toString()));
 
+    // todo:What should we do if version variable overflows?
     int nextVersion = (current.first() != null ? current.first() : 0) + 1;
     Path finalMetadataFile = metadataFilePath(nextVersion, codec);
     FileSystem fs = getFileSystem(tempMetadataFile, conf);
+    boolean versionCommitSuccess = false;
+    try{
+      // this rename operation is the atomic commit operation
+      renameToFinal(fs, tempMetadataFile, finalMetadataFile, nextVersion);
 
-    // this rename operation is the atomic commit operation
-    renameToFinal(fs, tempMetadataFile, finalMetadataFile, nextVersion);
+      LOG.info("Committed a new metadata file {}", finalMetadataFile);
 
-    LOG.info("Committed a new metadata file {}", finalMetadataFile);
+      // update the best-effort version pointer
+      versionCommitSuccess = writeVersionHint(nextVersion);
 
-    // update the best-effort version pointer
-    writeVersionHint(nextVersion);
+      deleteRemovedMetadataFiles(base, metadata);
 
-    deleteRemovedMetadataFiles(base, metadata);
-
-    this.shouldRefresh = true;
+      this.shouldRefresh = true;
+    }catch (Throwable e){
+      // If the versionHint has been submitted successfully, then we don't need to clean up the data file if the task fails.
```

Comment:
Not sure I understand this, "this.shouldRefresh" should only be able to be false here because if we got to the line this.shouldRefresh = true there is basically no way to throw an exception at that point.

Reply:
Since we are using SPARK-LLAP to execute ICEBERG-Query, we realise that every step of the code may throw a runtime exception. What if we throw a runtime exception when we execute deleteRemovedMetadataFiles?